plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -278,10 +282,15 @@ export class MySQLInstrumentation extends InstrumentationBase<
           values = [_valuesOrCallback];
         }
 
-        span.setAttribute(
-          SemanticAttributes.DB_STATEMENT,
-          getDbStatement(query, format, values)
-        );
+        const dbStatementSerializer =
+          typeof thisPlugin._config.dbStatementSerializer === 'function'
+            ? thisPlugin._config.dbStatementSerializer
+            : thisPlugin._defaultDbStatementSerializer.bind(thisPlugin);
+
+        let dbStatement = getDbStatement(query, format, values);
```

Comment:
The `getDbStatement` function actually [makes an effort](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/28ad01e040f44a7b9ff1f76c4ba486074e8faf9c/plugins/node/opentelemetry-instrumentation-mysql/src/utils.ts#L89) to insert these values which are not there in the first place:
```js
  if (typeof query === 'string') {
    return values ? format(query, values) : query;
  } else {
    // According to https://github.com/mysqljs/mysql#performing-queries
    // The values argument will override the values in the option object.
    return values || query.values
      ? format(query.sql, values || query.values)
      : query.sql;
  }
```

I wonder if we can just skip the `format` by default, and apply it only if we want to collect full payload. This will make it simpler and more efficient

Reply:
You can even consider capturing the values as another attributes like [in postgres](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/28ad01e040f44a7b9ff1f76c4ba486074e8faf9c/plugins/node/opentelemetry-instrumentation-pg/src/utils.ts#L166):

```js
  if (
    instrumentationConfig.enhancedDatabaseReporting &&
    Array.isArray(queryConfig.values)
  ) {
    span.setAttribute(
      AttributeNames.PG_VALUES,
      arrayStringifyHelper(queryConfig.values)
    );
  }
```

Might make it easier to process downstream in the spans pipeline, as the values are formatted and not embedded in the statement text